arduino/smartcar/smartcar.ino

```diff
@@ -155,7 +164,68 @@ void loop()
     
 }
 
-//If the connection breaks, this method will be called
+
+void stopZoneAutoBreak(long frontUltDis, long frontIRDis, long backIRDis)
+{
+     if (frontUltDis <= 100 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0 || backIRDis <= 40 && backIRDis != 0)//stop zone
+         {
+            if (canDrive)//check whether you're in the stop zone
+            {
+                car.setSpeed(0);
+                Serial.println("Emergency stop 1");
+            }
+            canDrive = false;//so the car can move in the stop soon
+        } else {
+            canDrive = true;//so the car will stop again if it hits the stop zone
+        }
+}
+
+void incomingAvoidanceThreshold(long frontUltDis, long frontIRDis, long backIRDis)
+{
+    if (frontUltDis <= 30 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0)//Ford threshold 1
+    {
+        car.setSpeed(0);
+        car.setSpeed(-90);
+        Serial.println("backing up level 1");
+        activeAvoidance = true;
+    } else if (frontUltDis <= 60 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0)//Ford threshold 2
+    {
+        car.setSpeed(0);
+        car.setSpeed(-60);
+        Serial.println("backing up level 2");
+        activeAvoidance = true;
+    } else if (frontUltDis <= 90 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0)//Ford threshold 3this
```

Comment:
At the end of you comment here you wrote 3this. Not sure what you wanted to write here. 
Add more details like i mentioned in the previous comment

Reply:
It's fixed now